pymagine/edge_detection.py

```diff
@@ -24,4 +27,30 @@ def edge_detection(image, color="Greys"):
   -------
   image 
   image returned with desired edge detection color filter applied
-  """
\ No newline at end of file
+    """
+    if not isinstance(filename, str):
+        raise TypeError("Image file path must be a string.")
+    
+    if not filename.endswith((".png", ".jpeg", ".jpg")):
+        raise TypeError("Image format must be png, jpg, or jpeg.")
+    
+    if filename.startswith(("https:", "http:", "www.")):
+        raise TypeError("Image file path can't be a URL, provide a local file path.")
+        
+    if color == "Greys" or color == "Purples" or color == "Blues" or color == "Greens" or color == "Oranges" or color == "Reds":
+        True
+    else:
+        raise ValueError("Color can only be one of: 'Greys', 'Purples', 'Blues', 'Greens', 'Oranges', 'Reds'")
+    
+    filt = np.ones((3,3))/8
+
+    if is_grey:
+        img = plt.imread(filename)
+        I_filt = convolve2d(img,filt, boundary='symm', mode='same')
+
+    else:
+        img = plt.imread(filename)
+        img = rgb2gray(img)
+        I_filt = convolve2d(img,filt, boundary='symm', mode='same')
+        
+    return plt.imshow(I_filt, cmap = color)
```

Comment:
Write image to the user's working directory
```suggestion
    cv2.imwrite("edge_detect.jpeg", I_filt)
```